.github/workflows/tracegen.yml

```diff
@@ -0,0 +1,60 @@
+name: tracegen
+on:
+  push:
+    branches: [ main ]
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+  pull_request:
```

Comment:
Should this run on pull requests as well? If so, does it push to the `latest` tag? That would not be appropriate.

Reply:
the job for pull requests only builds the `tracegen` tool so that we can make sure changes on that won't break it.